src/finput.js

```diff
@@ -325,11 +312,14 @@ class Finput {
       default:
         // all printable characters have a key with length of 1 
         // if a character has got this far it is an invalid character
-        if(e.key.length === 1 && !e.ctrlKey){
-          this.options.invalidKeyCallback(this.createInvalidKeyInfo(keyInfo));
-          e.preventDefault();
-        }
-        return;
+        const isInvalid = e.key.length === 1 && !e.ctrlKey;
+        keyInfo.valid = !isInvalid;
+    }
+
+    if (!keyInfo.valid) {
```

Comment:
Should be an easy enough change to the keyHandlers to make them return `keyInfo` (it's not just `valid` that's read from later on) so that we're using proper function return values rather than relying on side effects.